File: src/main/java/com/ithuangma/java/ai/langchain4j/Service/UserProfileService.java
package com.ithuangma.java.ai.langchain4j.Service;

import com.ithuangma.java.ai.langchain4j.entity.UserProfile;
import java.util.List;
import java.util.Map;

public interface UserProfileService {
    // 创建或更新用户画像
    UserProfile saveUserProfile(UserProfile userProfile);
    
    // 根据用户ID获取用户画像
    UserProfile getUserProfileByUserId(Long userId);
    
    // 根据用户ID列表批量获取用户画像
    List<UserProfile> getUserProfilesByUserIds(List<Long> userIds);
    
    // 更新用户标签
    UserProfile updateUserTags(Long userId, String tags);
    
    // 更新用户MBTI类型
    UserProfile updateMbtiType(Long userId, String mbtiType);
    
    // 根据画像特征匹配相似用户
    List<UserProfile> findSimilarUsers(Long userId, int limit);
    
    // 根据画像特征匹配适合的队伍
    List<Long> findSuitableTeamsByProfile(Long userId);
}

File: frontend/src/api/userProfileService.js
// src/api/userProfileService.js
import apiClient from './httpClient'

/**
 * 获取伙伴画像
 */
export const getUserProfile = async (userId) => {
    const response = await apiClient.get(`/api/v1/user-profile/${userId}`)
    return response
}

/**
 * 保存或更新伙伴画像
 */
export const saveUserProfile = async (userProfile) => {
    const response = await apiClient.post('/api/v1/user-profile', userProfile)
    return response
}

/**
 * 批量获取伙伴画像
 */
export const getUserProfilesBatch = async (userIds) => {
    const response = await apiClient.post('/api/v1/user-profile/batch', userIds)
    return response
}

/**
 * 更新用户标签
 */
export const updateUserTags = async (userId, tags) => {
    const response = await apiClient.put(`/api/v1/user-profile/${userId}/tags`, tags)
    return response
}

/**
 * 更新用户MBTI类型
 */
export const updateMbtiType = async (userId, mbtiType) => {
    const response = await apiClient.put(`/api/v1/user-profile/${userId}/mbti?mbtiType=${mbtiType}`)
    return response
}

/**
 * 查找相似用户
 */
export const findSimilarUsers = async (userId, limit = 5) => {
    const response = await apiClient.get(`/api/v1/user-profile/${userId}/similar-users?limit=${limit}`)
    return response
}

/**
 * 根据画像查找适合的队伍
 */
export const findSuitableTeams = async (userId) => {
    const response = await apiClient.get(`/api/v1/user-profile/${userId}/suitable-teams`)
    return response
}


/**
 * 上传头像
 */
export const uploadAvatar = async (file) => {
    const formData = new FormData()
    formData.append('file', file)
    
    const response = await apiClient.post('/admin/common/upload', formData, {
        headers: {
            'Content-Type': 'multipart/form-data'
        }
    })
    
    return response
}

/**
 * 更新用户头像
 */
export const updateUserAvatar = async (userId, avatarUrl) => {
    const response = await apiClient.put(`/api/v1/user-profile/${userId}/avatar`, { avatarUrl })
    return response
}